packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -196,6 +202,7 @@ export abstract class FluidDataStoreContext extends EventEmitter implements
         public readonly isLocalDataStore: boolean,
         bindChannel: (channel: IFluidDataStoreChannel) => void,
         protected pkg?: readonly string[],
+        protected _isRootDataStore: boolean = false,
```

Comment:
Can it be non-optional? Is the only reason it's optional - being last? If so, can you move it up?

Reply:
To clarify, I'm not sure we want to have defaults like that. Loading old file - sure, but runtime calling this API in same layer - not so much

Reply:
I made it optional so old files don't break. But I think you are right, runtime will call this in the same layer. When loading from old snapshots, we will provide a value based on whether we have the "isRootDataStore" property in the snapshot or not. So yeah, I can make this non-optional.